### PR TITLE
add merge exclude tests -- use adapter zone test

### LINKED ISSUE #410: [CT-1884] [Bug] add adapter tests for `merge_exclude_columns`
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

we're missing tests!

### Expected Behavior

we should test `merge_exclude_columns` on Snowflake in the Adapter Zone!

### Steps To Reproduce

it's a Bug of Omission

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt-core:
- dbt-snowflake:
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dave-connors-3**
resolves #5260 (dbt-core issue)
closes #410

### Description

This PR adds integration tests for the associated logic change in dbt-core [PR #5457](https://github.com/dbt-labs/dbt-core/pull/5457)

### Related PRs:
- https://github.com/dbt-labs/dbt-core/pull/6700
- https://github.com/dbt-labs/dbt-bigquery/pull/227
- https://github.com/dbt-labs/dbt-snowflake/pull/206
- https://github.com/dbt-labs/dbt-spark/pull/601

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-bigquery next" section.

### CODE REVIEW

**Review by VersusFacit — APPROVED** [review]
LGTM (barring the changeback on the dev requirements once we complete hashing out the adapter zone class)

File: tests/integration/simple_copy_test/test_simple_copy.py [hunk 1]
```diff
@@ -308,4 +308,4 @@ def test__snowflake__incremental_merge_columns(self):
             }
         })
         self.seed_and_run()
-        self.assertTablesEqual("incremental_update_cols", "expected_result")
+        self.assertTablesEqual("incremental_update_cols", "expected_result")
```
Comment (nssalian) [inline tests/integration/simple_copy_test/test_simple_copy.py:311]: this change might not be needed in this PR, worth excluding since it isn't related.

Comment (VersusFacit) [reply tests/integration/simple_copy_test/test_simple_copy.py:311]: Yeah it's just some whitespace that got nixed by black.

**Review by nssalian — APPROVED** [review]

File: .changes/unreleased/Fixes-20230123-133059.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+kind: Fixes
+body: add adapter zone test for merge_exlcude_columns
```
Comment (nssalian) [inline .changes/unreleased/Fixes-20230123-133059.yaml:2]: typo `merge_exlcude_columns`

**Pull request merged**